lib/src/baseFormValidation.ts

```diff
@@ -13,35 +16,64 @@ interface FormValidation {
   isValidationInProgress(): boolean;
   isFormDirty(): boolean;
   isFormPristine(): boolean;
-  addFieldValidation(key: string, validation: (value: string, vm: any) => FieldValidationResult, filter?: any): FormValidation;
-  addFieldValidationAsync(key: string, validation: (value: string, vm: any) => Promise<FieldValidationResult>, filter?: any): FormValidation;
 }
 
 export class BaseFormValidation implements FormValidation {
   private validationEngine: ValidationEngine;
+
   constructor(validationConstraints: ValidationConstraints) {
     this.validationEngine = new ValidationEngine();
     this.parseValidationConstraints(validationConstraints);
   }
 
   private parseValidationConstraints(validationConstraints: ValidationConstraints) {
     if (validationConstraints && typeof validationConstraints === 'object') {
-      if (validationConstraints.global && validationConstraints.global instanceof Array) {
-        this.addFormValidationFunctions(validationConstraints.global);
+      const { global, fields } = validationConstraints;
+      if (global && global instanceof Array) {
+        this.parseFormValidations(global);
+      }
+      if (fields && typeof fields === 'object') {
+        this.parseAllFieldsValidations(fields);
       }
     }
   }
 
-  private addFormValidationFunctions(validationFunctions: FormValidationFunction[]) {
+  private parseFieldValidations(constraint: string, fieldValidationConstraints: FieldValidationConstraint[]) {
+    if (fieldValidationConstraints instanceof Array) {
+      fieldValidationConstraints.forEach((fieldValidationConstraint) => {
+        if (fieldValidationConstraint && typeof fieldValidationConstraint === 'object') {
+          this.addFieldValidation(constraint, fieldValidationConstraint);
+        }
+      });
+    }
+  }
+
+  private parseAllFieldsValidations(fields: { [key: string]: FieldValidationConstraint[] }) {
+    for (let field in fields) {
+      this.parseFieldValidations(field, fields[field]);
+    }
+  }
+
+  private parseFormValidations(validationFunctions: FormValidationFunction[]) {
```

Comment:
I prefer order methods like we are using:

```
if (global && global instanceof Array) {
        this.parseFormValidations(global);
      }
      if (fields && typeof fields === 'object') {
        this.parseAllFieldsValidations(fields);
      }
...

  private parseFormValidations(validationFunctions: FormValidationFunction[]) {
   .....
  }
  private parseAllFieldsValidations(fields: { [key: string]: FieldValidationConstraint[] }) {
     .....
  }

  private parseFieldValidations(constraint: string, fieldValidationConstraints: FieldValidationConstraint[]) {
   ...
  }
```

So, you don't have to scroll down to search what are doing for example `parseFormValidations`